swap/protocol_test.go

```diff
@@ -210,17 +210,6 @@ func TestEmitCheque(t *testing.T) {
 	if creditorSwap.balances[debitor.ID()] != 0 {
 		t.Fatalf("Expected debitor balance to have been reset to %d, but it is %d", 0, creditorSwap.balances[debitor.ID()])
 	}
-	/*
-			TODO: This test actually fails now, because the two Swaps create independent backends,
-			thus when handling the cheque, it will actually complain (check ERROR log output)
-			with `error="no contract code at given address"`.
-			Therefore, the `lastReceivedCheque` is not being saved, and this check would fail.
-			So TODO is to find out how to address this (should be by having same backend when creating the Swap)
-		if creditorSwap.loadLastReceivedCheque(debitor.ID()) != cheque {
-			t.Fatalf("Expected exactly one cheque at creditor, but there are %d:", len(creditorSwap.cheques))
-		}
-	*/
-
```

Comment:
This needs to remain there. I added it, the test actually only checks that the balance is reset, but not that the cheque arrived.

Reply:
@holisticode , I removed it and created a PR for it instead. Isn't it clearer when we keep all TODOs as issues? Obviously, if you want it back I will do so!

Reply:
Created an issue AND put the TODO back in the code with a reference to the issue

Reply:
You did not put it back into the code though. 

I personally don't think it's a good idea to put links to issues into the code. I preferred to have the code back instead of the issue number.

In order to not block finishing up the PR I will merge this nevertheless.

Reply:
I put it back.